mc-server-wrapper-lib/src/parse.rs

```diff
@@ -64,15 +64,22 @@ impl ConsoleMsgSpecific {
             ConsoleMsgSpecific::PlayerAuth { name, uuid }
         } else if console_msg.msg_type == ConsoleMsgType::Info
             && (console_msg.thread_name.starts_with("Async Chat Thread")
-                || console_msg.msg.starts_with('<') && console_msg.thread_name == "Server thread")
+                || console_msg.msg.starts_with('<')
+                || console_msg.msg.starts_with("[Not Secure] <")
+                    && console_msg.thread_name == "Server thread")
         {
             let (name, msg) = {
-                let (name, remain) = console_msg
+                let (mut name, remain) = console_msg
                     .msg
                     // If a > cannot be found, this is not a player message
                     // and therefore we return
                     .split_at(console_msg.msg.find('>')?);
 
+                // trim "[Not Secure] " from player's name
+                if name.starts_with('[') {
+                    name = &name[13..];
+                }
+
```

Comment:
What do you think about doing this instead:

```rust
let (name, msg) = {
    let msg = console_msg
        .msg
        .strip_prefix("[Not Secure] ")
        .unwrap_or(&console_msg.msg);

    let (name, remain) = msg
        // If a > cannot be found, this is not a player message
        // and therefore we return
        .split_at(msg.find('>')?);

    // Trim "<" from the player's name and "> " from the msg
    (name[1..].to_string(), remain[2..].to_string())
};
```

I like how `strip_prefix(...)` makes things a bit clearer. (I should probably change more of the code here to use it.)

Reply:
😄 it's a handy method!